Sources/JSONParameterEncoder.swift

```diff
@@ -0,0 +1,16 @@
+import Foundation
+
+public struct JSONParameterEncoder: ParameterEncoder {
+  public func encode(urlRequest: inout URLRequest, with parameters: Parameters) throws {
+    do {
+      let jsonAsData = try JSONSerialization.data(withJSONObject: parameters,
+                                                  options: .prettyPrinted)
+      urlRequest.httpBody = jsonAsData
+      if urlRequest.value(forHTTPHeaderField: "Content-Type") == nil {
```

Comment:
Should be fine to just set the content type regardless.

Reply:
I'd personally avoid overwriting regardless, but I'm not married to it either. So I'll take the subtle suggestion.